Lib/asyncio/proactor_events.py

```diff
@@ -22,6 +23,24 @@
 from .log import logger
 
 
+def _set_socket_extra(transport, sock):
+    transport._extra['socket'] = sock
+
+    try:
+        transport._extra['sockname'] = sock.getsockname()
+    except (socket.error, AttributeError):
```

Comment:
When `AttributeError` is raised?
The only attribute lookup is `sock.getsockname()`.
`socket.socket` instance always has this method, isn't it?

Reply:
This and the `sock.getpeername()` were copied directly from the existing code: https://github.com/python/cpython/pull/1067/files#diff-b96109897b7f8f8ee1ec8f3104103dd6L453

Someone, I forget who, requested that i move this code block out to a common function since it was repeated a couple times in this file. So, I copy/pasted as it was exactly. I can remove the `AttributeError` if you'd like, I just want to make sure it won't break anything existing.